src/plugins/discover/public/utils/get_sharing_data.ts

```diff
@@ -96,7 +96,10 @@ export async function getSharingData(
        */
       const useFieldsApi = !config.get(SEARCH_FIELDS_FROM_SOURCE);
       if (useFieldsApi && columns.length) {
-        searchSource.setField('fields', columns);
+        searchSource.setField(
+          'fields',
+          columns.map((field) => ({ field, include_unmapped: 'true' }))
+        );
```

Comment:
QQ: so it would just fetch unmapped fields in case there were columns selected, but and this would resolved the issue 👍 
But shouldn't also those fields be fetched when no column is selected, to align with Discover?
![Bildschirmfoto 2022-05-31 um 09 09 17](https://user-images.githubusercontent.com/463851/171114013-b5ac35a5-001b-4c42-9d17-e0ecbdc06a96.png)

Reply:
@kertal Those will also be fetched. In this case, we receive `{ field: '*', include_unmapped: 'true' }` on the reporting side. I've covered that in the functional tests as part of this PR.

Reply:
Thank you!